test/suites/snapshot.robot

```diff
@@ -0,0 +1,88 @@
+Library     ../../.venv/lib/python3.10/site-packages/robot/libraries/Collections.py
+
+Resource    ../resources/microshift-process.resource
+Resource    ../resources/common.resource
+Resource    ../resources/kubeconfig.resource
+Resource    ../resources/oc.resource
+Resource    ../resources/microshift-config.resource
+
+Suite Setup         Test Suite Setup
+Suite Teardown      Teardown Suite With Namespace
+
+
+*** Variables ***
+${POD_NAME_STATIC}      base
+${TEST_DATA}            FOOBAR
+${SOURCE_KUSTOMIZE}     ./test/assets/kustomizations/patches/pvc-thin
+${RESTORE_KUSTOMIZE}    ./test/assets/kustomizations/patches/pvc-from-snapshot
+${SNAPSHOT}    ./test/assets/snapshot.yaml
+${STORAGE_CLASS}   ./test/assets/storage-class-thin.yaml
+${SNAPSHOT_CLASS}   ./test/assets/volume-snapshot-class.yaml
+
+*** Test Cases ***
+Snapshotter Smoke Test
+    [Documentation]     Write data to a volume, snapshot it, restore the snapshot and verify the data is present
+    [Tags]      smoke   snapshot
+    [Setup]     Test Case Setup
+    [Teardown]  Test Case Teardown
+
+    Oc Apply    -f ${SNAPSHOT} -n ${NAMESPACE}
+    Oc Wait For     volumesnapshot/my-snap      jsonpath\='{.status.readyToUse}=true'
+    Oc Apply    -k ${RESTORE_KUSTOMIZE} -n ${NAMESPACE}
+    Oc Wait For    pod/${POD_NAME_STATIC}    condition\=Ready
+    ${data}=    Read From Volume    ${POD_NAME_STATIC}
+    Should Be Equal As Strings    FOOBAR    ${data}
+
+
+*** Keywords ***
+Test Suite Setup
+    [Documentation]
+    Setup Suite With Namespace
+    Extend LVMD Config
+    Restart Microshift
+
+Test Case Setup
+    Oc Apply    -f ${STORAGE_CLASS} -f ${SNAPSHOT_CLASS}
+    Oc Apply    -k ${SOURCE_KUSTOMIZE} -n ${NAMESPACE}
+    Oc Wait For     pod/${POD_NAME_STATIC}    condition\=Ready
+    Write To Volume    ${POD_NAME_STATIC}    ${TEST_DATA}
+    Oc Delete   pod ${POD_NAME_STATIC} -n ${NAMESPACE}
+    Oc Wait For     pod/${POD_NAME_STATIC}      delete
+
+Test Case Teardown
+    Oc Delete   -f ${STORAGE_CLASS} -f ${SNAPSHOT_CLASS}
+
+Write To Volume
+    [Arguments]     ${to_pod}   ${data}
+    ${output}=      Oc Exec     ${to_pod}   echo "${data}" > /vol/data
+    Log     ${output}
+
+Read From Volume
+    [Arguments]     ${from_pod}
+    ${output}=      Oc Exec    ${from_pod}    cat /vol/data
+    RETURN      ${output}
+
+Extend LVMD Config
+    [Documentation]     Configure the lvmd file to
+    [Arguments]
+    # Get the existing lvmd from the cluster. This is preferable since no default lvmd.yaml is written to disk.
+    # And if one is written to disk, it will have been populated into the configMap already.
+    ${cm}=   Oc Get    cm    openshift-storage    lvmd
+    ${default_cfg}=  Set Variable  ${cm['data']['lvmd.yaml']}
+    # Load the lvmd config patch
+    ${patch_cfg}=   OperatingSystem.Get File    ./test/assets/lvmd.patch.yaml
+
+    ${merged_cfg}=     Lvmd Merge    ${default_cfg}    ${patch_cfg}
+    ${stdout}   ${stderr}     ${rc}=     Execute Command
```

Comment:
If this is writing the file to the microshift host filesystem, take a look at the keywords in the config resource file to see how that does something similar using file copying.

Reply:
I completely missed that 😅 .  Using the resource's keyword instead now.